R/derive_locf_records.R

```diff
@@ -0,0 +1,184 @@
+#'   "CDISC01", "01-701-1015", "SYSBP", "Systolic Blood Pressure (mmHg)", 121,    4,       "WEEK 4",
+#'   "CDISC01", "01-701-1015", "SYSBP", "Systolic Blood Pressure (mmHg)", 121,    6,       "WEEK 6",
+#'   "CDISC01", "01-701-1028", "DIABP", "Diastolic Blood Pressure (mmHg)", 79,    0,       "BASELINE",
+#'   "CDISC01", "01-701-1028", "DIABP", "Diastolic Blood Pressure (mmHg)", 80,    2,       "WEEK 2",
+#'   "CDISC01", "01-701-1028", "DIABP", "Diastolic Blood Pressure (mmHg)", NA,    4,       "WEEK 4",
+#'   "CDISC01", "01-701-1028", "DIABP", "Diastolic Blood Pressure (mmHg)", NA,    6,       "WEEK 6",
+#'   "CDISC01", "01-701-1028", "SYSBP", "Systolic Blood Pressure (mmHg)", 130,    0,       "BASELINE",
+#'   "CDISC01", "01-701-1028", "SYSBP", "Systolic Blood Pressure (mmHg)", 132,    2,       "WEEK 2"
+#' )
+#'
+#'
+#' # A dataset with all the combinations of PARAMCD, PARAM, AVISIT, AVISITN, ... which are expected.
+#' advs_expected_obsv <- tibble::tribble(
+#'  ~PARAMCD, ~PARAM, ~AVISITN, ~AVISIT,
+#'   "DIABP", "Diastolic Blood Pressure (mmHg)", 0, "BASELINE",
+#'   "DIABP", "Diastolic Blood Pressure (mmHg)", 2, "WEEK 2",
+#'   "DIABP", "Diastolic Blood Pressure (mmHg)", 4, "WEEK 4",
+#'   "DIABP", "Diastolic Blood Pressure (mmHg)", 6, "WEEK 6",
+#'   "SYSBP", "Systolic Blood Pressure (mmHg)",  0, "BASELINE",
+#'   "SYSBP", "Systolic Blood Pressure (mmHg)",  2, "WEEK 2",
+#'   "SYSBP", "Systolic Blood Pressure (mmHg)",  4, "WEEK 4",
+#'   "SYSBP", "Systolic Blood Pressure (mmHg)",  6, "WEEK 6"
+#' )
+#'
+#' # Add a new record for each 'by_vars' when the input dataset does not contain
+#' # observations for missed visits/timepoints or when AVAL is missing.
+#' # Set DTYPE = 'LOCF' for the newly added records.
+#' # Carry forward the AVAL of the previous observation after sorting the dataset
+#' # by 'order' parameter.
+#' Final_advs <- derive_locf_records(data=advs,
+#'                                   dataset_expected_obs=advs_expected_obsv,
+#'                                   by_vars=vars(STUDYID, USUBJID, PARAM, PARAMCD),
+#'                                   order=vars(STUDYID, USUBJID, PARAM, PARAMCD, AVISITN, AVISIT))
+
+
+
+derive_locf_records <- function(dataset,
+                                dataset_expected_obs,
+                                by_vars,
+                                order) {
+  library(tidyr)
+
+  #### Input Checking ####
+
+  # Check by_vars and order variables in input dataset
+  assert_data_frame(dataset, required_vars = by_vars)
+  assert_data_frame(dataset, required_vars = order)
+
+  # Check if input parameters is a valid list of variables
+  assert_vars(by_vars, optional = TRUE)
+  assert_order_vars(order)
+
+
+
+  #### Prepping 'dataset_expected_obs' ####
+
+  # Add temp_dtype='LOCF' for the 'expected observation dataset'
+  dataset_expected_obs <- dataset_expected_obs %>%
+    mutate(temp_dtype="LOCF")
```

Comment:
Please use [get_new_tmp_var()](https://pharmaverse.github.io/admiraldev/devel/reference/get_new_tmp_var.html) to create temporary variables.
I would move it to line 156. Then it is close to the place where it is used and you do not need to remove it in line 141.